### PR TITLE
feat(ct): navigate with page.goto

### LINKED ISSUE #15889: [Feature] Component testing and page.goto
**Context:**
- Playwright Version: 1.24.0
- Operating System: All
- Node.js version:  v17.3.0
- Browser: Chromium
- Extra: Vue component testing

I've been playing with the Vue component testing feature and it seems like the `page.goto()` is not working properly.

For example 
[i have a failing test that should be passing](https://github.com/sand4rt/TVmaze/blob/master/src/Features/NotFound/NotFoundPage.test.ts).

It should navigate to the 404 page because i have a catch all specified [`/:pathMatch(.*)*`](https://router.vuejs.org/guide/essentials/dynamic-matching.html#catch-all-404-not-found-route) inside Vue router [see the code here](https://github.com/sand4rt/TVmaze/blob/master/src/Router/Index.ts#L52).

The Vue router plugin is registered [here](https://github.com/sand4rt/TVmaze/blob/master/playwright/index.ts).

[Here](https://github.com/sand4rt/TVmaze/blob/master/src/Features/Browse/BrowsePage.test.ts#L61) you can see another example that also uses the `page.goto` function that i can't get to work.

### PR DESCRIPTION
**Pull request by sand4rt**
closes: https://github.com/microsoft/playwright/issues/15889

### CODE REVIEW

File: packages/playwright-test/src/mount.ts [hunk 1]
```diff
@@ -48,23 +48,27 @@ export const fixtures: Fixtures<
...
+                await window.playwrightUnmount(rootElement);
+              });
+            },
+            update: async (options: JsxComponent | Omit<MountOptions, 'hooksConfig'>) => {
+              if (isJsxApi(options)) return await innerUpdate(page, options);
+              await innerUpdate(page, component, options);
+            }
+          });
+        }
+        page.addListener('load', mount);
```
Comment (pavelfeldman) [inline packages/playwright-test/src/mount.ts:70]: This change suggests that it is Ok to navigate in the component tests. But in reality, only 'good' navigations to a magic URL will result in the harness result (the one with window.playwrightMount available. If we want to support it, we should ensure that the page navigated to any URL allows mounting?

Comment (sand4rt) [reply packages/playwright-test/src/mount.ts:70]: > If we want to support

I have some router logic which i think is useful to test with `page.goto` and `page.refresh`: https://github.com/sand4rt/vue-router-test/blob/master/src/router/index.ts#L6

Also wrote some tests for this logic: https://github.com/sand4rt/vue-router-test/blob/master/src/notation-vue.test.ts#L4

Wonder what you think and if there is a better way to test this logic?

> We should ensure that the page navigated to any URL allows mounting?

When testing components, navigating to pages that don't start with the base URL `process.env.PLAYWRIGHT_TEST_BASE_URL` seems not useful to me. A URL that starts with the base URL but doesn't exist/is not registered in [`router/index.ts`](https://github.com/sand4rt/vue-router-test/blob/master/src/router/index.ts#L8), like: `http://localhost:3100/page-that-does-not-exists` is in my opinion useful to keep.

File: tests/components/ct-vue-vite/src/notation-jsx.spec.tsx [hunk 2]
```diff
@@ -143,11 +143,20 @@ test('get textContent of the empty template', async ({ mount }) => {
...
   await component.getByRole('link', { name: 'Dashboard' }).click();
   await expect(component.getByRole('main')).toHaveText('Dashboard');
   await expect(page).toHaveURL('/dashboard');
 });
+
+test('navigate to a page with page.goto', async ({ page, mount }) => {
+  const component = await mount(<App />);
+  await expect(component.getByRole('main')).toHaveText('Login');
+  await expect(page).toHaveURL('/');
+  await page.goto('/dashboard');
```
Comment (pavelfeldman) [inline tests/components/ct-vue-vite/src/notation-jsx.spec.tsx:159]: I'm not sure I understand why App is mounted in /dashboard after the navigation, this looks confusing to me.

**Comment by pavelfeldman** [comment]
Let me push back to this one until I understand the use case better.

**Pull request closed without merging**